packages/rest/src/lib/handlers/SequentialHandler.ts

```diff
@@ -304,11 +304,12 @@ export class SequentialHandler {
 		try {
 			// node-fetch typings are a bit weird, so we have to cast to any to get the correct signature
 			// Type 'AbortSignal' is not assignable to type 'import("discord.js-modules/node_modules/@types/node-fetch/externals").AbortSignal'
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 			res = await fetch(url, { ...options, signal: controller.signal as any });
 		} catch (error: unknown) {
 			// Retry the specified number of times for possible timed out requests
 			if (error instanceof Error && error.name === 'AbortError' && retries !== this.manager.options.retries) {
-				return this.runRequest(routeId, url, options, bodyData, ++retries);
+				return await this.runRequest(routeId, url, options, bodyData, ++retries);
```

Comment:
By adding this await, the finally block (line 317 in this PR) will be called later, causing intervals to accumulate without clearing between recursive calls.

What exactly is the ESLint error that required this to be added?

Reply:
https://typescript-eslint.io/rules/return-await/

Reply:
Not actually sure this matters in this particular instance because to reach that code I'm fairly certain the timeout has already finished, causing the AbortError